cmake/GNU.cmake

```diff
@@ -0,0 +1,58 @@
+
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fcray-pointer -ffree-line-length-none -fno-range-check -fbacktrace")
+
+if(DEBUG)
+    message("DEBUG is       ENABLED")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -Wall -O0 -ggdb -fno-unsafe-math-optimizations -frounding-math -fsignaling-nans -ffpe-trap=invalid,zero,overflow -fbounds-check")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -O0 -ggdb")
+else()
+    message("DEBUG is       disabled (optimized build)")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O2 -fno-range-check")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2")
+endif()
+
+
+if(REPRO)
+    message("REPRO is       ENABLED")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O2 -ggdb")
```

Comment:
I found that on macOS, you need to change `-O2` to `-O0` to get b4b identical results between IPD and CCPP if (and only if) the dynamics are compiled in 32bit (independent of whether gcc+gfortran or clang+gfortran are used). I would suggest to have another test inside the `if(REPRO)` block that tests for macOS (I believe `if(APPLE)` works) that uses `-O0` instead of `-O2`. For Linux, this problem doesn't exist.

Reply:
I can not test on macOS. Do you want to make required changes?

Reply:
Will do, yes.

Reply:
Committed in d62d01a